docs/en/optimizing-query-objects.md

```diff
@@ -56,12 +56,14 @@ class ArticlesQuery extends QueryObject
 
   public function postFetch(Queryable $repository, \Iterator $iterator)
   {
-    $ids = array_keys(iterator_to_array($iterator, TRUE));
```

Comment:
Tiis didn't work for me beacuse the entities were not indexed by their ids. Do I have to enable it somehow or is it not possible?